src/pyobs/aop.py

```diff
@@ -419,6 +477,122 @@ def getAOP(self,what,Species=None,wavelength=None):
 
         raise AOPError("not implemented yet")
 
+    def getPM(self,Species=None,wavelength=None,pmsize=None,fixrh=None,aerodynamic=False):
+        """
+        Returns an xarray Dataset with particulate matter smaller than the size given.
+
+        Species:  None, str, or list. If None, all species on file,
+                  otherwise subset of emissions.
+                  
+        Wavelength: float, wavelength in nm. 
+    
+        PMsize: float, particle diameter threshold in microns. If None, the total PM is calculated.
+
+        """
+
+        # All species on file or a subset
+        # -------------------------------
+        if Species is None:
+            Species = list(self.mieTable.keys())
+        if isinstance(Species,str):
+            Species = [Species,]
+
+        a = self.aer    # aerosol mixing ratio tracers
+
+        # Determine PM Threshold
+        # -------------------------------
+
+        if pmsize is None:
+            pmsize = 99999 #A ridiculous value is chosen such that the threshold is larger than any aerosol particle
+        rPM = float(pmsize)/2 #convert diameter to radius
+
+        # GEOS files can be inconsistent when it comes to case
+        # ----------------------------------------------------
+        try:
+            dp = a['DELP'] 
+        except:
+            dp = a['delp']
+
+        rh = a['RH']
+
+        # Check FIXRH option
+        # --------------------------
+        if fixrh is not None:
+            fixrh = float(fixrh)
+            print('Fixed RH = ',fixrh)
+            if(fixrh >= 0.):
+                if(fixrh > 1.):
+                   print("Your --RH > 1, must be between 0 - 1; exit and fix")
+                   exit()
+                rh = rh*0.0+fixrh
+
+        # Relevant dimensions
+        # -------------------
+        space = rh.shape
+
+        pm = (np.zeros(space))
+        for s in Species:   # species
+
+            if self.verbose:
+                print('[] working on',s)
+            
+            Tracers = self.mieTable[s]['tracers']
+            mie = self.mieTable[s]['mie']
+
+            bin = 1
+            for q in Tracers:
+
+                if self.verbose:
+                    print('   -',q)
+
+                
+                q_conc = (a['AIRDENS'] * a[q]).values * 1000000000 #convert units to kg/m3
+                #rhod is not in all of the standard optics files, and is temporarily added in the yaml portion of the code. 
+                #rhod_ = mie.getAOP('rhod',  bin, rh, wavelength=wavelength).values
+                rhod_ = self.mieTable[s]['rhod'][bin-1]
```

Comment:
add some comments that explain what each of these variables is, and why you're scaling by 1e6